libs/intro/generate/helpers/slides.ts

```diff
@@ -0,0 +1,52 @@
+import { readJSON } from 'fs-extra';
+import { google } from 'googleapis';
+import { OAuth2Client } from 'googleapis-common';
+import { forkJoin, from, ObservableInput } from 'rxjs';
+import {
+  catchError,
+  map,
+  switchMap,
+  switchMapTo,
+  tap,
+  withLatestFrom
+} from 'rxjs/operators';
+
+import { GSlides, Token } from './types';
+import { CODELAB_PRESENTATION_ID, TOKEN_PATH } from './const';
+import { oAuth2Client$, getNewToken$ } from './utils';
+
+export const fetchSlides$ = oAuth2Client$.pipe(
+  map((auth: OAuth2Client): GSlides => google.slides({ version: 'v1', auth })),
```

Comment:
This file doesn't need to know anything about the auth, it only needs gSlides$ and pipe through it